src/catalog/src/information_schema/partitions.rs

```diff
@@ -0,0 +1,405 @@
+use common_recordbatch::adapter::RecordBatchStreamAdapter;
+use common_recordbatch::{RecordBatch, SendableRecordBatchStream};
+use common_time::datetime::DateTime;
+use datafusion::physical_plan::stream::RecordBatchStreamAdapter as DfRecordBatchStreamAdapter;
+use datafusion::physical_plan::streaming::PartitionStream as DfPartitionStream;
+use datafusion::physical_plan::SendableRecordBatchStream as DfSendableRecordBatchStream;
+use datatypes::prelude::{ConcreteDataType, ScalarVectorBuilder, VectorRef};
+use datatypes::schema::{ColumnSchema, Schema, SchemaRef};
+use datatypes::value::Value;
+use datatypes::vectors::{
+    ConstantVector, DateTimeVector, DateTimeVectorBuilder, Int64Vector, Int64VectorBuilder,
+    MutableVector, StringVector, StringVectorBuilder, UInt64VectorBuilder,
+};
+use futures::TryStreamExt;
+use partition::manager::PartitionInfo;
+use partition::partition::PartitionDef;
+use snafu::{OptionExt, ResultExt};
+use store_api::storage::{RegionId, ScanRequest, TableId};
+use table::metadata::{TableInfo, TableType};
+
+use super::PARTITIONS;
+use crate::error::{
+    CreateRecordBatchSnafu, FindPartitionsSnafu, InternalSnafu, Result,
+    UpgradeWeakCatalogManagerRefSnafu,
+};
+use crate::information_schema::{InformationTable, Predicates};
+use crate::kvbackend::KvBackendCatalogManager;
+use crate::CatalogManager;
+
+const TABLE_CATALOG: &str = "table_catalog";
+const TABLE_SCHEMA: &str = "table_schema";
+const TABLE_NAME: &str = "table_name";
+const PARTITION_NAME: &str = "partition_name";
+const PARTITION_EXPRESSION: &str = "partition_expression";
+/// The region id
+const GREPTIME_PARTITION_ID: &str = "greptime_partition_id";
+
+/// The `PARTITIONS` table provides information about partitioned tables.
+/// See https://dev.mysql.com/doc/refman/8.0/en/information-schema-partitions-table.html
+/// We provide an extral column `greptime_partition_id` for GreptimeDB region id.
+pub(super) struct InformationSchemaPartitions {
+    schema: SchemaRef,
+    catalog_name: String,
+    catalog_manager: Weak<dyn CatalogManager>,
+}
+
+impl InformationSchemaPartitions {
+    pub(super) fn new(catalog_name: String, catalog_manager: Weak<dyn CatalogManager>) -> Self {
+        Self {
+            schema: Self::schema(),
+            catalog_name,
+            catalog_manager,
+        }
+    }
+
+    pub(crate) fn schema() -> SchemaRef {
+        Arc::new(Schema::new(vec![
+            ColumnSchema::new(TABLE_CATALOG, ConcreteDataType::string_datatype(), false),
+            ColumnSchema::new(TABLE_SCHEMA, ConcreteDataType::string_datatype(), false),
+            ColumnSchema::new(TABLE_NAME, ConcreteDataType::string_datatype(), false),
+            ColumnSchema::new(PARTITION_NAME, ConcreteDataType::string_datatype(), false),
+            ColumnSchema::new(
+                "subpartition_name",
+                ConcreteDataType::string_datatype(),
+                true,
+            ),
+            ColumnSchema::new(
```

Comment:
Why are some column names defined as const and some as literal values?

Reply:
All the column names not defined as const are `null` values for compatibility, they are useless.